options.c

```diff
@@ -1368,6 +1371,27 @@ static void toggle_block_key_paste(void *data)
 	block_key_paste ^= 1;
 }
 
+const char * const progress_bar_names[] = {
+	"disabled", "line", "shuttle", "color", "color_shuttle", NULL
+};
+
+static void get_progress_bar(void *data, char *buf, size_t size)
+{
+	strscpy(buf, progress_bar_names[progress_bar], size);
+}
+
+static void set_progress_bar(void *data, const char *buf)
+{
+	parse_enum(buf, 0, 4, progress_bar_names, &progress_bar);
+}
+
+static void toggle_progress_bar(void *data)
+{
+	progress_bar++;
+	progress_bar %= 5;
```

Comment:
Something like `NR_PROGRESS_BAR_TYPES` would be good to avoid hard-coding 5. Also, `cycle` (or similar) naming might be more appropriate than `toggle`.

Reply:
Agree but currently it's only exposed as ":toggle" (or win-toggle/spacebar in the settings screen) and all matching settings use the toggle_* naming. I'm open to adding a :cycle alias but I think it's different from what you were asking.

(And oops, missed the rebase in my first push.)